yoti-sdk-impl/src/main/java/com/yoti/api/client/docs/DocScanClient.java

```diff
@@ -0,0 +1,128 @@
+package com.yoti.api.client.docs;
+
+import com.yoti.api.client.InitialisationException;
+import com.yoti.api.client.KeyPairSource;
+import com.yoti.api.client.Media;
+import com.yoti.api.client.docs.session.create.CreateSessionResult;
+import com.yoti.api.client.docs.session.create.SessionSpec;
+import com.yoti.api.client.docs.session.retrieve.DocScanSession;
+import com.yoti.api.client.spi.remote.KeyStreamVisitor;
+import com.yoti.api.client.spi.remote.call.docs.DocScanService;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.KeyPair;
+import java.security.Security;
+
+/**
+ * Client used for communication with the Yoti Doc Scan service,
+ * where any signed request is required.
+ * <p>
+ * The {@code DocScanClient} is to be used by clients to facilitate
+ * requests to the Yoti Doc Scan system where any signed requests are
+ * required.  Using the supplied models, clients can build requests
+ * in native Java, and perform the requests to the Doc Scan system.
+ *
+ * @since 2.7
+ */
+public class DocScanClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DocScanClient.class);
+
+    static {
+        Security.addProvider(new BouncyCastleProvider());
+    }
+
+    private final String sdkId;
+    private final KeyPair keyPair;
+
+    private final DocScanService docScanService;
+
+    DocScanClient(final String sdkId,
+                  final KeyPairSource keyPairSource,
+                  DocScanService docScanService) {
+        this.sdkId = sdkId;
```

Comment:
Assuming we can't enforce the DocScanClient being initialised through the builder, do you think we should add validation for the parameters here?

Reply:
If we want to make it testable, not really.  We have default access on it which means someone would have to access the class from within the same package, whereas the builder has public access so it's kind of implied...

Reply:
Sorry, didn't realise it was internal/package-private, this is fine then